src/metabase/api/card.clj

```diff
@@ -174,24 +174,29 @@
 
 ;;; -------------------------------------------------- Saving Cards --------------------------------------------------
 
-;; When a new Card is saved, we wouldn't normally have the results metadata for it until the first time its query is
-;; ran.  As a workaround, we'll calculate this metadata and return it with all query responses, and then let the
-;; frontend pass it back to us when saving or updating a Card.  As a basic step to make sure the Metadata is valid
-;; we'll also pass a simple checksum and have the frontend pass it back to us.  See the QP `results-metadata`
-;; middleware namespace for more details
-
 (s/defn ^:private result-metadata-async :- ManyToManyChannel
-  "Get the right results metadata for this `card`, and return them in a channel. We'll check to see whether the
-  `metadata` passed in seems valid,and, if so, return a channel that returns the value as-is; otherwise, we'll run the
-  query ourselves to get the right values, and return a channel where you can listen for the results."
-  [query metadata]
-  (let [valid-metadata? (s/validate qr/ResultsMetadata metadata)]
-    (log/info
-     (if valid-metadata?
-       (trs "Card results metadata passed in to API is VALID. Thanks!")
-       (trs "Card results metadata passed in to API is MISSING. Running query to fetch correct metadata.")))
-    (if valid-metadata?
+  "Return a channel of metadata for the passed in `query`. Takes the `original-query` so it can determine if existing
```

Comment:
Not sure if this is still relevant but I don't think these functions actually ought to be async. I took a crack at making this synchronous way back in #14807 but never got it over the line

Reply:
With all the changes going on in metadata, I'd rather a transparent refactor change land in 42 rather than making all of these changes harder to understand.